### PR TITLE
UP19-4 View Inventory

### PR DESCRIPTION
**Pull request by murphyad**
Link to story: https://msoese.atlassian.net/browse/UP19-4

- Added page to view organization's inventory
- Items on page display items:
    - Tracking ID (name)
    - Note (description)
    - Quantity w/ units
    - Date Created
    - Image (or default if src is not provided or invalid)
- Org catalog cards now have a button to redirect user to the inventory of that org
- Page styled similarly to the org catalog.

### CODE REVIEW

File: src/pages/OrganizationCatalog.js [hunk 1]
```diff
@@ -1,30 +1,46 @@
...
-                <p >{isNullOrUndefined(agent.primaryLocation) ? "(distance not available)" : "Distance: "+getDistanceBetweenPoints(agent.primaryLocation, msoeCC).toFixed(2)+" mi"}</p>
-            </Item.Description>
-            <Item.Extra>
-                <Button className="ui right floated primary">Connect</Button>
-            </Item.Extra>
-        </Item.Content>
-    </Item>
-)
+    viewInventory = () => {
+        console.log(this.props);
```
Comment (larsonme) [inline src/pages/OrganizationCatalog.js]: Remove this printout

File: src/pages/OrganizationCatalog.js [hunk 2]
```diff
@@ -1,30 +1,46 @@
...
+            <Item.Content>
+                <Item.Header as='h1'>{this.props.agent.name}</Item.Header>
+
+                <Item.Description>
+                    <p>{isNullOrUndefined(this.props.agent.primaryLocation) ? "(no location available)" : this.props.agent.primaryLocation.address}</p>
+                    <p>{isNullOrUndefined(this.props.agent.primaryLocation) ? "(distance not available)" : "Distance: " + getDistanceBetweenPoints(this.props.agent.primaryLocation, msoeCC).toFixed(2) + " mi"}</p>
+                </Item.Description>
+                <Item.Extra>
+                    <Button className="ui right floated primary" onClick={this.viewInventory}>View
+                        Inventory</Button>
```
Comment (larsonme) [inline src/pages/OrganizationCatalog.js]: Throw this all on one line

File: src/pages/OrganizationCatalog.js [hunk 3]
```diff
@@ -149,7 +165,7 @@ class BasePage extends React.Component {
                     filteredOrgs = sortByDistance(filteredOrgs, msoeCC);
                 }
                 const cardsArray = filteredOrgs.map(org => (
-                    orgCard(org)
+                    <OrgCard agent={org} history={this.props.history}/>
```
Comment (larsonme) [inline src/pages/OrganizationCatalog.js]: Rename agent prop to org.  More specific and better readability for groups going forwards

File: src/pages/Organizations/organizationInventory.js [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+let default_image = require("../../resources/default_resource_img.jpg");
+
+/**
+ * Gets an organizations data
+ */
+export const GetSingleOrganization = getOrganizationById(({ organization, loading, error }) => {
+    if (loading) {
+        return(
+            <strong>Loading...</strong>
```
Comment (larsonme) [inline src/pages/Organizations/organizationInventory.js]: Replace these with Semantic UI messages(I use a loading on in the header bar)

File: src/pages/OrganizationCatalog.js (hunk 2 above)
Comment (StrikeNP) [reply src/pages/OrganizationCatalog.js]: I'm fine with 2 lines, but 'view' and 'inventory' should be on one line at least.

File: src/pages/Organizations/organizationInventory.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        );
+    } else if (error) {
+        return(
+            <p style={{color: "#F00"}}>API error</p>
+        );
+    } else {
+        let economicResourceList = organization.ownedEconomicResources;
+        console.log(economicResourceList.length === 0);
```
Comment (larsonme) [inline src/pages/Organizations/organizationInventory.js]: Remove print

File: src/pages/Organizations/organizationInventory.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        );
+    }
+});
+
+/**
+ * Gets a single economic resource that the organization owns
+ */
+export const GetSingleEconomicResource = getEconomicResourceById(({ economicResource, loading, error }) => {
+    if (loading) {
+        return(
```
Comment (larsonme) [inline src/pages/Organizations/organizationInventory.js:48]: Same with replacing with semantic ui loading component

File: src/queries/EconomicResource/economicResourceInterface.tsx [hunk 7]
```diff
@@ -29,10 +29,12 @@ fragment economicResourceInterface on EconomicResource {
   transfers{
     id
   }
+  createdDate
 }`;
 
 export interface EconomicResource {
   id: number
+  createDate: string
   resourceClassifiedAs: number // TODO Replace with ResourceClassification
```
Comment (DonalMoloneyMSOE) [inline src/queries/EconomicResource/economicResourceInterface.tsx:38]: possibly remove Todo

**Review by larsonme — APPROVED** [review]

**Review by DonalMoloneyMSOE — APPROVED** [review]

File: src/pages/Organizations/organizationInventory.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+            <strong>Loading...</strong>
```
Comment (StrikeNP) [inline src/pages/Organizations/organizationInventory.js]: Same as Michael's comment above, can we use the semantic ui loading here instead?

File: src/pages/Organizations/organizationInventory.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        );
+    } else {
+        let economicResourceList = organization.ownedEconomicResources;
+        console.log(economicResourceList.length === 0);
+        return(
+            <div>
+                <h2 className="ui header">{organization.name} Inventory</h2>
+                <Item.Group divided>
+                {(economicResourceList.length ===0)? <p>Inventory Empty</p> :(economicResourceList.map( (economicResource) =>
+                    (<GetSingleEconomicResource economicResourceId={economicResource.id}/>)
```
Comment (StrikeNP) [inline src/pages/Organizations/organizationInventory.js:35]: This is an unusually large a mount of logic in our return. Is it possible to move this outside of here? If not, it's probably ok 'enough' to leave it.

**Review by StrikeNP — APPROVED** [review]

**Pull request merged**